collector/writer.go

```diff
@@ -29,56 +32,108 @@ func (w CollectdWriter) Write(s Stats) error {
+		"memory-usage": s.Stats.MemoryStats.Usage,
+
+		"memory-active_anon":   s.Stats.MemoryStats.Stats.TotalActiveAnon,
+		"memory-active_file":   s.Stats.MemoryStats.Stats.TotalActiveFile,
+		"memory-cache":         s.Stats.MemoryStats.Stats.TotalCache,
+		"memory-inactive_anon": s.Stats.MemoryStats.Stats.TotalInactiveAnon,
+		"memory-inactive_file": s.Stats.MemoryStats.Stats.TotalInactiveFile,
+		"memory-mapped_file":   s.Stats.MemoryStats.Stats.TotalMappedFile,
+		"memory-pg_fault":      s.Stats.MemoryStats.Stats.TotalPgfault,
+		"memory-pg_in":         s.Stats.MemoryStats.Stats.TotalPgpgin,
+		"memory-pg_out":        s.Stats.MemoryStats.Stats.TotalPgpgout,
+		"memory-rss":           s.Stats.MemoryStats.Stats.TotalRss,
+		"memory-rss_huge":      s.Stats.MemoryStats.Stats.TotalRssHuge,
+		"memory-unevictable":   s.Stats.MemoryStats.Stats.TotalUnevictable,
+		"memory-writeback":     s.Stats.MemoryStats.Stats.TotalWriteback,
+	}
+
+	network_metrics := map[string]string{}
+
 	for _, network := range s.Stats.Networks {
-		metrics["net.rx_bytes"] += network.RxBytes
-		metrics["net.rx_dropped"] += network.RxDropped
-		metrics["net.rx_errors"] += network.RxErrors
-		metrics["net.rx_packets"] += network.RxPackets
-
-		metrics["net.tx_bytes"] += network.TxBytes
-		metrics["net.tx_dropped"] += network.TxDropped
-		metrics["net.tx_errors"] += network.TxErrors
-		metrics["net.tx_packets"] += network.TxPackets
+		network_metrics["if_dropped"] = fmt.Sprintf("%d:%d", network.RxDropped, network.TxDropped)
+		network_metrics["if_octets"] = fmt.Sprintf("%d:%d", network.RxBytes, network.TxBytes)
+		network_metrics["if_errors"] = fmt.Sprintf("%d:%d", network.RxErrors, network.TxErrors)
+		network_metrics["if_packets"] = fmt.Sprintf("%d:%d", network.RxPackets, network.TxPackets)
 	}
 
 	t := s.Stats.Read.Unix()
 
-	for k, v := range metrics {
+	for k, v := range cpu_metrics {
+		err := w.writeFloat(s, k, t, v)
+		if err != nil {
+			return err
+		}
+	}
+
+	for k, v := range memory_metrics {
 		err := w.writeInt(s, k, t, v)
 		if err != nil {
 			return err
 		}
 	}
 
+	for k, v := range network_metrics {
+		err := w.writeString(s, k, t, v)
+		if err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+func (w CollectdWriter) writeFloat(s Stats, k string, t int64, v float64) error {
+	msg := fmt.Sprintf(collectdFloatGaugeTemplate, w.host, s.App, s.Task, k, t, v)
+	_, err := w.writer.Write([]byte(msg))
+	return err
+}
+
 func (w CollectdWriter) writeInt(s Stats, k string, t int64, v uint64) error {
 	msg := fmt.Sprintf(collectdIntGaugeTemplate, w.host, s.App, s.Task, k, t, v)
 	_, err := w.writer.Write([]byte(msg))
 	return err
 }
+
+func (w CollectdWriter) writeString(s Stats, k string, t int64, v string) error {
+	msg := fmt.Sprintf(collectdStringTemplate, w.host, s.App, s.Task, k, t, v)
+	_, err := w.writer.Write([]byte(msg))
+	return err
+}
+
+func calculateCPUPercent(previous docker.CPUStats, current docker.CPUStats) (float64, float64, float64) {
```

Comment:
@joshgarnett this is not percentage, there is cpu time for this. Instead of doing percentage you just need to do rate of cpu time. Zero new metrics required for this.